### PR TITLE
Merge tag 'swift-DEVELOPMENT-SNAPSHOT-2019-09-24-a' into tensorflow

### PR DESCRIPTION
**Pull request by dan-zheng**
Recreation of https://github.com/apple/swift/pull/27397, merging from `apple/tensorflow-merge` branch.

### CODE REVIEW

**Review by burmako — APPROVED** [review]
LGTM after the merge conflict is fixed and tests pass.

**Comment by dan-zheng** [comment]
@swift-ci Please test tensorflow

**Comment by dan-zheng** [comment]
@swift-ci Please clean test tensorflow

**Pull request merged**
